x/staking/keeper/liquid_stake_test.go

```diff
@@ -509,3 +510,172 @@ func TestDecreaseValidatorTotalLiquidShares(t *testing.T) {
+
+	// The first account should now have tokenize shares disabled
+	// and the unlock time should be the zero time
+	disabled, actualUnlockTime := app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressA)
+	require.True(t, disabled, "addressA locked")
+	require.True(t, actualUnlockTime.IsZero(), "addressA unlock time uninititalized")
+
+	disabled, _ = app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressB)
+	require.False(t, disabled, "addressB still unlocked")
+
+	// Update the lock time and confirm it was set
+	expectedUnlockTime := time.Date(2023, 1, 1, 0, 0, 0, 0, time.UTC)
+	app.StakingKeeper.SetTokenizeShareUnlockTime(ctx, addressA, expectedUnlockTime)
+
+	disabled, actualUnlockTime = app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressA)
+	require.True(t, disabled, "addressA still locked")
+	require.Equal(t, expectedUnlockTime, actualUnlockTime, "addressA unlock time")
+
+	// Confirm B is still unlocked
+	disabled, _ = app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressB)
+	require.False(t, disabled, "addressB still unlocked")
+
+	// Remove the lock
+	app.StakingKeeper.RemoveTokenizeSharesLock(ctx, addressA)
+	disabled, _ = app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressA)
+	require.False(t, disabled, "addressA unlocked at end")
+
+	disabled, _ = app.StakingKeeper.IsTokenizeSharesDisabled(ctx, addressB)
+	require.False(t, disabled, "addressB unlocked at end")
+}
+
+// Test Get/SetPendingTokenizeShareAuthorizations
+func TestPendingTokenizeShareAuthorizations(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	// Create dummy accounts and completion times
+	addresses := simapp.AddTestAddrs(app, ctx, 3, sdk.NewInt(1))
+	addressStrings := []string{}
+	for _, address := range addresses {
+		addressStrings = append(addressStrings, address.String())
+	}
+
+	timeA := time.Date(2023, 1, 1, 0, 0, 0, 0, time.UTC)
+	timeB := timeA.Add(time.Hour)
+
+	// There should be no addresses returned originally
+	authorizationsA := app.StakingKeeper.GetPendingTokenizeShareAuthorizations(ctx, timeA)
+	require.Empty(t, authorizationsA.Addresses, "no addresses at timeA expected")
+
+	authorizationsB := app.StakingKeeper.GetPendingTokenizeShareAuthorizations(ctx, timeB)
+	require.Empty(t, authorizationsB.Addresses, "no addresses at timeB expected")
+
+	// Store addresses for timeB
+	app.StakingKeeper.SetPendingTokenizeShareAuthorizations(ctx, timeB, types.PendingTokenizeShareAuthorizations{
+		Addresses: addressStrings,
+	})
+
+	// Check addresses
+	authorizationsA = app.StakingKeeper.GetPendingTokenizeShareAuthorizations(ctx, timeA)
+	require.Empty(t, authorizationsA.Addresses, "no addresses at timeA expected at end")
+
+	authorizationsB = app.StakingKeeper.GetPendingTokenizeShareAuthorizations(ctx, timeB)
+	require.Equal(t, addressStrings, authorizationsB.Addresses, "address length")
+}
+
+// Test QueueTokenizeSharesAuthorization and RemoveExpiredTokenizeShareLocks
+func TestTokenizeShareAuthorizationQueue(t *testing.T) {
```

Comment:
Can you confirm that there is no stochasticity to queues in golang? We've run into trouble with new data structures (eg maps) before so I'm wary

Reply:
sweet, let's keep it